src/Kafka/ManageKafkaPermissions/components/SelectAccount.tsx

```diff
@@ -0,0 +1,206 @@
+};
+
+export const SelectAccount: React.VFC<SelectAccountProps> = ({
+  onChangeAccount,
+  id,
+  accounts,
+  onEscapeModal,
+}) => {
+  const { t } = useTranslation(["manage-kafka-permissions"]);
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const [validated, setValidated] = useState<ValidatedOptions>(
+    ValidatedOptions.default
+  );
+
+  const onToggle = (newState: boolean | ((prevState: boolean) => boolean)) => {
+    if (newState) {
+      onEscapeModal(false);
+    } else onEscapeModal(true);
+
+    setIsOpen(newState);
+  };
+
+  const clearSelection = () => {
+    onChangeAccount(undefined);
+    setValidated(ValidatedOptions.error);
+    setIsOpen(false);
+  };
+  const serviceAccountOptions = () => {
+    const serviceAccountsLength = accounts.filter(
+      (account) => account.principalType === PrincipalType.ServiceAccount
+    ).length;
+    return serviceAccountsLength
+      ? accounts
+          .filter(
+            (principal) =>
+              principal.principalType === PrincipalType.ServiceAccount
+          )
+          .sort((a, b) =>
+            a.displayName && b.displayName
+              ? a.displayName.localeCompare(b.displayName)
+              : -1
+          )
+          .map((principal, index) => (
+            <SelectOption
+              key={index}
+              value={principal.id}
+              description={principal.displayName}
+            >
+              {principal.id}
+            </SelectOption>
+          ))
+      : [
+          <SelectOption isNoResultsOption={true} isDisabled={true} key={1}>
+            {t("manage_permissions_dialog.no_results_found")}
+          </SelectOption>,
+        ];
+  };
+
+  const userAccountOperations = () => {
+    const userAccountsLength = accounts.filter(
+      (account) => account.principalType === PrincipalType.UserAccount
+    ).length;
+    return userAccountsLength
+      ? accounts
+          .filter(
+            (principal) => principal.principalType === PrincipalType.UserAccount
+          )
+          .map((principal, index) => (
+            <SelectOption
+              key={index}
+              value={principal.id}
+              description={principal.displayName}
+            >
+              {principal.id}
+            </SelectOption>
+          ))
+      : [
+          <SelectOption isNoResultsOption={true} isDisabled={true} key={1}>
+            {t("manage_permissions_dialog.no_results_found")}
+          </SelectOption>,
+        ];
+  };
+
+  const onSelect: SelectProps["onSelect"] = (_, selection) => {
+    onChangeAccount(selection);
+    setValidated(ValidatedOptions.default);
+    setIsOpen(false);
+  };
+  const options = [
+    <SelectGroup key="all_accounts_group">
+      <SelectOption
+        key="*"
+        value="*"
+        description={t("manage_permissions_dialog.all_accounts_description")}
+      >
+        {t("manage_permissions_dialog.all_accounts_title")}
+      </SelectOption>
```

Comment:
I don't think the "All accounts" option should be inside a `<SelectGroup>` component. Since this section of the menu doesn't have a group label, including `<SelectGroup>` without a `label` prop results in extra space above this option, making it look inconsistent with other menus. 

<img width="455" alt="image" src="https://user-images.githubusercontent.com/21063328/156415182-ff85bf61-3cea-4d1b-8f62-c4ee72438a70.png">